cpp/src/Ice/SSL/SSLUtil.cpp

```diff
@@ -97,8 +97,8 @@ Ice::SSL::readFile(const string& file, vector<char>& buffer)
     }
 }
 
-bool
-Ice::SSL::checkPath(const string& path, const string& defaultDir, bool dir, string& resolved)
+optional<string>
+Ice::SSL::checkPath(const string& path, const string& defaultDir, bool dir)
```

Comment:
I find the name "defaultDir" misleading. It sounds like it's the path we'd return upon failure, when it's really an optional parent directory for "path".

Do we need checkPath to perform this "parent dir" + path concatenation? If yes, I would rename defaultDir -> parentDir and make it an `optional<string>`.